packages/dd-trace/test/plugins/util/inferred_proxy.spec.js

```diff
@@ -37,14 +35,19 @@ describe('Inferred Proxy Spans', function () {
       }
     })
 
-    appListener = server.listen(port, '127.0.0.1')
+    return new Promise((resolve, reject) => {
+      appListener = server.listen(0, '127.0.0.1', () => {
+        port = server.address().port
+        resolve()
+      })
+    })
   }
 
   // test cleanup function
-  const cleanupTest = function () {
+  const cleanupTest = async function () {
```

Comment:
I just had a look at the tests themselves. If I am not mistaken, test failures would be silenced due to this cleanup being called in a catch handler. So instead, I think this should be a afterEach hook.